azurerm/provider.go

```diff
@@ -424,24 +430,23 @@ func determineAzureResourceProvidersToRegister(providerList []resources.Provider
 // whether they are actually used by the configuration or not). It was confirmed by Microsoft
 // that this is the approach their own internal tools also take.
 func registerAzureResourceProvidersWithSubscription(providerList []resources.Provider, client resources.ProvidersClient) error {
+	providers := determineAzureResourceProvidersToRegister(providerList)
+
 	var err error
-	providerRegistrationOnce.Do(func() {
```

Comment:
I'm not 100% sure of the original need for this - I'm assuming it's needed for the acceptance tests (which run in parallel) - to verify this I'll run the acceptance tests against this branch shortly to confirm

Reply:
Acceptance tests passed - so looks like we're good here :)

Reply:
Cool, good the hear that! And thanks for the review :+1: